prow/cmd/tackle/main.go

```diff
@@ -459,8 +459,8 @@ func githubClient(tokenPath string, dry bool) (*github.Client, error) {
 	return github.NewClient(gen, "https://api.github.com"), nil
 }
 
-func applySecret(ctx, name, key, path string) error {
-	return applyCreate(ctx, "secret", "generic", name, "--from-file="+key+"="+path)
+func applySecret(ctx, ns, name, key, path string) error {
+	return applyCreate(ctx, ns, "secret", "generic", name, "--from-file="+key+"="+path)
```

Comment:
How about
a) adding --namespace=default into this line
b) printing an error/warning when the context sets a context
c) then after that creating a plan to make tackle and starter.yaml allow you to deploy to a non-default namespace?

Reply:
Do you mean 
```
return applyCreate(ctx, "secret", "generic", name, "--from-file="+key+"="+path, "--namespace=default")
```
?

Also, not sure what you mean in b)

Reply:
Yes that's what I mean for a

For (b) I mean if I'm selecting a context that specifies a different default namespace, but our tooling forces the namespace to be default we should error or warn so the user isn't going to be surprised.